index.js

```diff
@@ -209,16 +209,16 @@ module.exports = (opts = {}) => {
       }
 
       return {
-        AtRule: {
-          'add-mixin': (node, helpers) => {
-            insertMixin(helpers, mixins, node, opts)
-          },
-          'define-mixin': (node, helpers) => {
-            addMixin(helpers, mixins, node)
-            node.remove()
-          },
-          'mixin': (node, helpers) => {
-            insertMixin(helpers, mixins, node, opts)
+        AtRule(node, helpers) {
+          switch (node.name) {
```

Comment:
It will slow down the plugin, because object structure is faster (callback will be called only on specific at-rule, not for every at-rule).

Reply:
I've run (a poor man's) benchmark on a simple scenario that includes different AtRule but to be honest I haven't seen any noticeable performance degradation.
```
  const times = [];

  for (let i = 0; i < 1000000; i++) {
    const before = performance.now();
    await run(
      '@mixin a { @supports (max(0px)) { color: black; } }',
      '.a { @supports (max(0px)) { color: black; } }',
      {
        mixins: {
          a() { return { '.a': { '@mixin-content' : {} } } }
        }
      })
    const after = performance.now();
    times.push(after - before);
  }
 ```
 
 ```
 Old:

Total: 75515.94069999667
Mean: 0.07551594069999668
Standard deviation: 0.025997863098952936

New:

Total: 76030.33139999444
Mean: 0.07603033139999443
Standard deviation: 0.026244211710887977
```

Reply:
You need to do the benchmark on big CSS with many different at-rules (`@apply`, `@media`, etc).

If your benchmark there is only one at-rule, so there is no useless callback calls.